File: EsBot-Server/Domain/Entities/UserSession.cs
using System.ComponentModel.DataAnnotations;

namespace Domain.Entities;

public record UserSession
{
    [Key] public Guid Id { get; set; }
    
    [Required] [MaxLength(100)] public string ExternalUserId { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime LastInteractionAt { get; set; } = DateTime.UtcNow;
    public ICollection<Message> Messages { get; set; } = new List<Message>();
    public ICollection<QuizRequest> QuizRequests { get; set; } = new List<QuizRequest>();
}

File: backend/app/models/user_session.py
from sqlalchemy import Column, Integer, DateTime
from sqlalchemy.orm import relationship
from datetime import datetime
from app.database import Base

class UserSession(Base):
    __tablename__ = "user_sessions"

    id = Column(Integer, primary_key=True, index=True)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)

    messages = relationship("Message", back_populates="session", cascade="all, delete-orphan")
    quiz_requests = relationship("QuizRequest", back_populates="session", cascade="all, delete-orphan")

    def add_message(self, message):
        message.session = self
        self.messages.append(message)

    def add_quiz_request(self, quiz_request):
        quiz_request.session = self
        self.quiz_requests.append(quiz_request)